File: vcl/AsyncClientServerSSL/Suite 9x/ServerSSL/clSslServer.pas
unit clSslServer;

interface

uses
  Classes, clTcpServer, clTcpServerTls, clUtils;

type
  TclSslServer = class(TclTcpServerTls)
  protected
    function CreateDefaultConnection: TclUserConnection; override;
    procedure DoReadConnection(AConnection: TclUserConnection; AData: TStream); override;
  end;

implementation

{ TclSslServer }

function TclSslServer.CreateDefaultConnection: TclUserConnection;
begin
  Result := TclUserConnectionTls.Create();
end;

procedure TclSslServer.DoReadConnection(AConnection: TclUserConnection; AData: TStream);
begin
  if (AData.Size = 0) then Exit;

  inherited DoReadConnection(AConnection, AData);

  //send data to the client
  AData.Position := 0;
  AConnection.WriteData(AData);
end;

end.

File: vcl/AsyncClientServerSSL/CBuilder/ServerSSL/clSslServer.cpp
//---------------------------------------------------------------------------


#pragma hdrstop

#include "clSslServer.h"

//---------------------------------------------------------------------------

#pragma package(smart_init)

void __fastcall TclSslServer::DoReadConnection(TclUserConnection* AConnection, Classes::TStream* AData)
{
  if (AData->Size == 0) return;

  TclTcpServer::DoReadConnection(AConnection, AData);

  //send data to the client
  AData->Position = 0;
  AConnection->WriteData(AData);
}

